### PR TITLE
Feature/redbox 200 move sources to a separate field for chat endpoint responses

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a frontend I want the backed to return as much information as possible., i.e. the uuid of the original file.

## Changes proposed in this pull request

* I have extended the `ChatResponse` to return the full langchain response
* I have extended the test to check that the original document is returned
* I have switched to the incremental test approach (thx @wpfl-dbt)

## Relevant links

https://technologyprogramme.atlassian.net/browse/REDBOX-200
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/8910714772/job/24470508092

### CODE REVIEW

File: django_app/redbox_app/redbox_core/migrations/0003_chathistory_file_chatmessage_and_more.py [hunk 1]
```diff
@@ -74,9 +73,7 @@ class Migration(migrations.Migration):
                 ("name", models.TextField(max_length=1024)),
                 (
                     "selected_files",
-                    models.ManyToManyField(
-                        blank=True, related_name="chat_histories", to="redbox_core.file"
-                    ),
+                    models.ManyToManyField(blank=True, related_name="chat_histories", to="redbox_core.file"),
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/migrations/0003_chathistory_file_chatmessage_and_more.py:76]: this is ruff modifying some unrelated files

File: core_api/src/routes/chat.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def simple_chat(chat_request: ChatRequest) -> ChatResponse:
     return ChatResponse(response_message=ChatMessage(text=response.text, role="ai"))
 
 
-@chat_app.post("/rag", tags=["chat"], response_model=ChatResponse)
+@chat_app.post("/rag", tags=["chat"])
```
Comment (lmwilkigov) [inline core_api/src/routes/chat.py:115]: Does the response model need to be returned?

**Review by lmwilkigov — APPROVED** [review]

File: core_api/src/routes/chat.py (hunk 2 above)
Comment (gecBurton) [reply core_api/src/routes/chat.py:115]: not if it is in the annotation (below). ofc its not doing any harm either, happy to out it back?

File: redbox/models/chat.py [hunk 3]
```diff
@@ -35,11 +36,32 @@ class ChatRequest(BaseModel):
...
+class InputDocuments(BaseModel):
+    page_content: str = Field(description="chunk text")
+    metadata: Metadata
+
+
+class ChatResponse(BaseModel):
+    question: str = Field(
+        description="original question",
+        examples=["Who is the prime minister?"],
+    )
```
[the remark this answers is not in the document]
Comment (gecBurton) [reply redbox/models/chat.py:60]: the frontend know all of the documents that could have been used in RAG, but not the documents actually used in RAG. The point of this change is that the document references come back in a helpful format rather than the `Doc12AC-0FE-8CB4-....` text format

[the remark this answers is not in the document]
Comment (gecBurton) [reply redbox/models/chat.py:60]: ahh, good point. yes lets drop it, its unhelpful

File: redbox/models/chat.py [hunk 4]
```diff
@@ -35,11 +36,32 @@ class ChatRequest(BaseModel):
...
+    input_documents: Optional[list[InputDocuments]] = Field(
```
[the remark this answers is not in the document]
Comment (gecBurton) [reply redbox/models/chat.py:61]: `sources` sounds good, it is what appears in the text afterall
